src/mainwindow/mainwindow.h

```diff
@@ -908,7 +908,7 @@ protected slots:
     bool m_restarting;
 
     QStringList m_alienFiles;
-    QStringList m_peAlienFiles;
+    QStringList peAlienFiles;
```

Comment:
Uhm... this is not the same variable... here it *is* a member variable of the class MainWindow, so the m_ prefix should be there.

Reply:
That should mean this declaration is in the wrong place then. What is happening is the parts editor is running in one thread and creates the parts files, the file names of which need to go in to m_alienFiles in the mainwindow thread so it knows to delete them if the user chooses to (the parts editor does not delete them because that is user choice later). Here that is achieved by a Qt ipc call (signal) to basically copy a local variable from one process to the other (which seems to me to be a violation of the OOP principles). So I appear to either have the define in src/mainwindow/mainwindow.h in the wrong place, or this is not the correct way to do what I want to do (although I think I copied this from something else that did much the same thing, but that is a year and a half ago at this point). The idea is the src/mainwindow/mainwindow.h file is included in both mainwindow.cpp and in pemainwindow.cpp so they both know what how the  local variable that has appeared via IPC is defined. It seemed to me it doesn't want to be public, so protected (while I don't really know what that means, I had assumed private to the module) the same as the current m_alienFiles seemed the best bet and the compiler appears happy with it as the code appears to work. Is there some other way to define peAlienFiles so it is defined in both pemainwindow.cpp and mainwindow.cpp, or perhaps is the correct answer to not do the IPC call at all but copy the file names in to m_alienFiles directly which would mean getting m_alienFiles passed in to pemainwindow.cpp from somewhere . I think the file names are destroyed just after the current signal call and the part is passed as a model structure without the required file names to the mainwindow function. This method works, but that doesn't mean it is the best way to do this.

Reply:
From a modeling point of view yes, the main window should not need to know details about parts, for example that parts include so alienFiles. 
How this is best realized with the existing structure I can not tell. Just the fact that a protected or private variable is modified via an IPC call is not a violation of OOP. But bloating implementation details into one giant class is.
There is already a lot of stuff in mainwindow (the sheer file size and number of members is quite a strong indicator that there is something wrong). But is it possible handle this in the parts editor window?

Reply:
"There is already a lot of stuff in mainwindow (the sheer file size and number of members is quite a strong indicator that there is something wrong). But is it possible handle this in the parts editor window?"

Unfortunately  no, while pemainwindow creates the files, it then puts the part in the mine parts bin without the file names being added to alienFiles (that may be where this should be done but I don't know how yet) . In mainwindow.cpp as Fritzing is exiting, it asks if you want to save the parts in the mine parts bin. If the user says no, then it uses the file names in alienFiles to delete the no longer needed files from the parts directory. Currently, (and what this fix tries to correct) the files for the part created by parts editor aren't in alienFiles and thus don't get deleted, which means you can'r reload that same part again because the files are present, but you can't delete the files from within Fritzing because the part no longer exists (only the files) and you need to manually remove the files from the parts directory outside of Fritzing. I made a push with the requested changes and an attempt at fixing this problem by moving the peAlienFiles definition to the top of the mainwindow.h file, but that doesn't compile as the symbol is multiply defined. I expect it needs to be extern in the .h file, and defined in one or another of the .cpp files but I'm having trouble figuring out which one it should be in (I think pemainwindow.cpp). I now have a version that compiles, but it doesn't seem to run correctly. I'll keep poking.